### DISCUSSION TITLE
The mask format is very confusing / not intuitive from examples

### DISCUSSION THREAD

**Discussion by matthew-dean** [beholdr/maska#174] [Q&A]
I've been a web dev for over 20 years, and worked with Regex for a good portion of that, and for the last hour, I haven't been able to make heads or tails of how masking characters and patterns are supposed to work.

For example, say I want something that's a very simple regex. That is:
1. I want to only allow alphanumeric characters
2. (Multiple) characters can be separated by single spaces.

In other words, it's a regex like: `/^[A-Za-z0-9]+(\s[A-Za-z0-9]+)*$/`

However, I can't figure out from the examples and lots of trial and error how to express this in the Maska format or using Maska options. It seems to match one token at a time, and that token is not preserved, meaning there's no logic about what comes after, say, a single space.

Similarly, I can't figure out how to match something like dash-case, where you can have only single dash separators, and no trailing or preceding dashes.

Can anyone help?

**Comment by matthew-dean** [comment]
Maybe none of the use cases I'm trying to do are supported? I can't understand why they wouldn't be, though, for a masked input. 🤔

**Reply by matthew-dean** [reply]
Looking through the examples, it looks like Maska isn't versatile enough for these scenarios. That's disappointing.

**Comment by albertcito** [comment]
It worked to me: 

```html
<input
      v-maska
      data-maska="A"
      data-maska-tokens="A:[-_ A-Za-z0-9]:multiple"
>
```
